instrumentation/testing/testing_test.go

```diff
@@ -0,0 +1,34 @@
+package testing
+
+import (
+	"strings"
+	"testing"
+)
+
+func TestLogBufferRegex(t *testing.T) {
+	test := StartTest(t)
+	defer test.End()
+
+	logLines := []string{
+		"Hello World",
+		"Hello        World     With         Spaces",
+		"Hello\n World\nMulti\n        Line",
+	}
+
+	for _, item := range logLines {
+		t.Log(item)
+	}
+
+	logBuffer := extractTestOutput(t)
+	logs := string(*logBuffer)
+	for idx, matches := range TESTING_LOG_REGEX.FindAllStringSubmatch(logs, -1) {
+		message := strings.ReplaceAll(matches[3], "\n        ", "\n")
+		if logLines[idx] != message {
+			t.FailNow()
+		}
+	}
```

Comment:
Instead of doing this, can we have an expected `logLinesExpected` array and test against that? If not, we are not really testing what we are expecting, as we are matching it against the same logic.

Reply:
- `logLines` is the expected array so I change the name to `expectedLogLines`.
- create a new func called `findMatchesLogRegex` that handles all the logic to extract the log lines (used by `extractTestLoggerOutput` ).
- change the test to use the new func, so we test the expected lines against the extraction logic.